### PR TITLE
ar_coordination module skeleton

### PR DESCRIPTION
**Pull request by ElFantasma**
Initial version of `ar_coordination` module

### CODE REVIEW

File: apps/arweave/src/ar_coordination.erl [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                    L when is_list(L) ->
+                        maps:put(PartitionId, [{Peer, PartitionSize, Packing}|L], Acc);
+                    none ->
+				        maps:put(PartitionId, [{Peer, PartitionSize, Packing}], Acc)
+                end
+			end,
+			State#state.mining_peers,
+            StorageModules
+		),
+    State#state{mining_peers = MiningPeers}.
```
Comment (AminArria) [inline apps/arweave/src/ar_coordination.erl]: nit: missing new line at end of file

File: apps/arweave/src/ar_coordination.erl [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+%%%===================================================================
+
+add_mining_peer({Peer, StorageModules}, State) ->
+    MiningPeers =
+        lists:foldl(
+			fun	({PartitionSize, PartitionId, Packing}, Acc) ->
+                %% Allowing the case of same partition handled by different peers
+                case maps:get(PartitionId, Acc, none) of
+                    L when is_list(L) ->
+                        maps:put(PartitionId, [{Peer, PartitionSize, Packing}|L], Acc);
```
Comment (AminArria) [inline apps/arweave/src/ar_coordination.erl]: We need a space before/after `|` separator according to arweave styleguide

**Review by AminArria — APPROVED** [review]

File: apps/arweave/src/ar_coordination.erl (hunk 2 above)
Comment (ElFantasma) [reply apps/arweave/src/ar_coordination.erl]: fixed!

File: apps/arweave/src/ar_coordination.erl (hunk 1 above)
Comment (ElFantasma) [reply apps/arweave/src/ar_coordination.erl]: fixed!

**Pull request merged**